pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/rebalance/RebalanceSegmentsFactory.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.linkedin.pinot.controller.helix.core.rebalance;
+
+import com.linkedin.pinot.common.config.TableConfig;
+import org.apache.helix.HelixManager;
+
+
+/**
+ * Singleton factory class, to fetch the right rebalance segments strategy based on table config
+ */
+public class RebalanceSegmentsFactory {
```

Comment:
Can we rename to `RebalanceSegmentStrategyFactory`?

Reply:
Fixed renaming and formatting